specs/phase1/validator.md

```diff
@@ -0,0 +1,529 @@
+                a for a in shard_attestations
+                if a.data.shard_transition_root == shard_transition_root
+            ]
+            transition_participants: Set[ValidatorIndex] = set()
+            for attestation in transition_attestations:
+                participants = get_attesting_indices(state, attestation.data, attestation.aggregation_bits)
+                transition_participants = transition_participants.union(participants)
+
+            enough_online_stake = (
+                get_total_balance(state, online_indices.intersection(transition_participants)) * 3 >=
+                get_total_balance(state, online_indices.intersection(committee)) * 2
+            )
+            if enough_online_stake:
+                shards.append(shard)
+                winning_roots.append(shard_transition_root)
+                break
+
+    return shards, winning_roots
+```
+
+##### Light client fields
+
+First retrieve `best_aggregate` from `get_best_light_client_aggregate` where `aggregates` is a list of valid aggregated `LightClientVote`s for the previous slot.
+
+Then:
+* Set `light_client_bits = best_aggregate.aggregation_bits`
+* Set `light_client_signature = best_aggregate.signature`
+
+```python
+def get_best_light_client_aggregate(block: BeaconBlock,
+                                    aggregates: Sequence[LightClientVote]) -> LightClientVote:
+    viable_aggregates = [
+        aggregate for aggregate in aggregates
+        if aggregate.slot == compute_previous_slot(block.slot) and aggregate.beacon_block_root == block.parent_root
+    ]
+
+    return max(
+        viable_aggregates,
+        key=lambda a: len([i for i in a.aggregation_bits if i == 1]),
+        default=LightClientVote(),
+    )
+```
+
+#### Packaging into a `SignedBeaconBlock`
+
+Packaging into a `SignedBeaconBlock` is unchanged from Phase 0.
+
+### Attesting
+
+A validator is expected to create, sign, and broadcast an attestation during each epoch.
+
+Assignments and the core of this duty are unchanged from Phase 0. There are a few additional fields related to the assigned shard chain and custody bit.
+
+The `Attestation` and `AttestationData` defined in the [Phase 1 Beacon Chain spec]() utilizes `shard_transition_root: Root` rather than a full `ShardTransition`. For the purposes of the validator and p2p layer, a modified `FullAttestationData` and containing `FullAttestation` are used to send the accompanying `ShardTransition` in its entirety. Note that due to the properties of SSZ `hash_tree_root`, the root and signatures of `AttestationData` and `FullAttestationData` are equivalent.
+
+#### `FullAttestationData`
+
+```python
+class FullAttestationData(Container):
+    slot: Slot
+    index: CommitteeIndex
+    # LMD GHOST vote
+    beacon_block_root: Root
+    # FFG vote
+    source: Checkpoint
+    target: Checkpoint
+    # Current-slot shard block root
+    shard_head_root: Root
+    # Full shard transition
+    shard_transition: ShardTransition
```

Comment:
I've thought about this a little bit. I think we might want to gossip the `ShardTransition` (or `AttestationData` separately so that message deduplication naturally happens

Will save for another PR